lib/routes/account.js

```diff
@@ -333,6 +348,196 @@ module.exports = function (
+                      uid: account.uid,
+                      email: account.email,
+                      emailCode: account.emailCode,
+                      emailVerified: true,
+                      verifierSetAt: account.verifierSetAt
+                    }
+                  )
+                  .then(
+                    function (sessionToken) {
+                      return db.createKeyFetchToken(
+                        {
+                          uid: account.uid,
+                          kA: account.kA,
+                          // wrapKb is undefined without a password
+                          // wrapWrapKb has the properties we need for this
+                          // value; Its stable, random, and will change on
+                          // account reset.
+                          wrapKb: account.wrapWrapKb,
+                          emailVerified: true
+                        }
+                      )
+                      .then(
+                        function (keyFetchToken) {
+                          return {
+                            sessionToken: sessionToken,
+                            keyFetchToken: keyFetchToken,
+                            unwrapBKey: butil.xorBuffers(
+                              account.kA,
+                              account.wrapWrapKb
+                            )
+                          }
+                        }
+                      )
+                    }
+                  )
+                  .then(
+                    function (tokens) {
+                      reply.redirect(
+                        config.contentServer.url + '/openid?' +
+                        qs.stringify(
+                          {
+                            uid: tokens.sessionToken.uid.toString('hex'),
+                            email: account.email,
+                            session: tokens.sessionToken.data.toString('hex'),
+                            key: tokens.keyFetchToken ?
+                              tokens.keyFetchToken.data.toString('hex')
+                              : undefined,
+                            unwrap: tokens.unwrapBKey.toString('hex'),
+                            service: 'sync',
+                            context: 'fx_desktop_v2'
+                          }
+                        )
+                      )
+                    }
+                  )
+                }
+              )
+              .catch(
+                function (err) {
+                  log.error({ op: 'Account.openid', err: err })
+                  reply.redirect(
+                    config.contentServer.url + '/openid?' + qs.stringify(
+                      {
+                        err: err.toString()
+                      }
+                    )
+                  )
+                }
+              )
+          },
+          false, // stateless
+          openIdExtensions,
+          false // strict
+        )
+      }
+    },
```

Comment:
factoring nit: I wonder if there's enough code here that we should pull these routes out into their own file, for a bit more separation between "core account" and "openid" stuff?